core/controllers/profile.py

```diff
@@ -480,6 +485,27 @@ def post(self):
 class SiteLanguageHandler(base.BaseHandler):
     """Changes the preferred system language in the user's preferences."""
 
+    # We want to impose regex to check whether the language
+    # code is valid. We choose the regex to be
+    # ^[A-Za-z]{2,4}([_-][A-Za-z]{4})?([_-]([A-Za-z]{2}|[0-9]{3}))?$
+    # This pattern can detect language code like "en", "en-US"
+    # "ah-Hant-TW", "EN-au".
+    regex_pat = '^[A-Za-z]{2,4}([_-][A-Za-z]{4})?([_-]([A-Za-z]{2}|[0-9]{3}))?$'
```

Comment:
How did you come up with this regex? How did you know which rules to use? Is there some kind of RFC or ISO for this thing?
```suggestion
    # We want to impose regex to check whether the language
    # code is valid. We choose the regex to be
    # ^[A-Za-z]{2,4}([_-][A-Za-z]{4})?([_-]([A-Za-z]{2}|[0-9]{3}))?$
    # This pattern can detect language code like "en", "en-US"
    # "ah-Hant-TW", "EN-au".
    regex_pat = '^[A-Za-z]{2,4}([_-][A-Za-z]{4})?([_-]([A-Za-z]{2}|[0-9]{3}))?$'
```

Reply:
@vojtechjelinek So the language code in `asset/constants.ts` consists of mostly `2` char to `3` char character codes and that is our base. 

However, the above regex could be more comprehensive in detecting other format also, including "en-US" and other types. I have tested locally that it can handle the language code listed in `asset/constants.ts`. Below is a snapshot for the language code

![Screen Shot 2021-10-04 at 10 35 43 PM](https://user-images.githubusercontent.com/72321026/135956128-4f569206-5c99-4224-8058-021b7ad2ee17.png)

Reply:
Also @vojtechjelinek the suggestion you provided above is identical?

Thanks!

Reply:
I get this but how do you know that 'en-US' is a valid type?

Reply:
@vojtechjelinek Never mind, I agree that there are too many types of language codes out there. I will modify the PR so that it will only conform to the format defined in the `asset/constants.ts` which is two lower case letters. For example `ta`, `en`, `es`. Please see screen shots below:

![Screen Shot 2021-10-05 at 11 49 59 AM](https://user-images.githubusercontent.com/72321026/136067296-36389116-8e8a-4fa2-a9bc-fd7c5a913c18.png)

Reply:
@vojtechjelinek Another idea is that we could have another function in `schema_utils.py` that call `is_valid_language_code(.....)`. The benefit would be we will not reply on regex and not only we will check whether it has the correct format, but also we can check that whether the content would be correct. 

For example, right now our regex will say that `wk` is correct, however `wk` has the right format but  is not a valid language code and should return false. 

Just an idea, we don't have to implement this and can just merge the PR with the regex check

Thanks

Reply:
@vojtechjelinek Yes, I think we can use this library, but do you think this is an overkill?